python/paddle/fluid/optimizer.py

```diff
@@ -4376,6 +4376,18 @@ def _create_vars(self, block, ori_block):
                         name=var,
                         type=core.VarDesc.VarType.READER,
                         persistable=source_var.persistable)
+                elif isinstance(source_var, Parameter):
+                    dest_var = block.create_parameter(
+                        name=source_var.name,
+                        shape=source_var.shape,
+                        dtype=source_var.dtype,
+                        type=source_var.type,
+                        lod_level=source_var.lod_level,
+                        stop_gradient=source_var.stop_gradient,
+                        trainable=source_var.trainable,
+                        optimize_attr=source_var.optimize_attr,
+                        regularizer=source_var.regularizer,
+                        error_clip=source_var.error_clip)
```

Comment:
We should also keep the `is_distributed` attribute for Parameter.

Reply:
in line 4393，we called `self._clone_var_attr()` which will copy `is_distributed` attr from `source_var` to `dest_var`